### PR TITLE
HAWQ-260. Resource manager wrongly calculates number of vseg

### PR DESCRIPTION
**Pull request by jiny2**
This is a bug.

### CODE REVIEW

**Comment by huor** [comment]
The fix looks good. +1

File: src/backend/resourcemanager/resqueuemanager.c [hunk 1]
```diff
@@ -3384,9 +3384,8 @@ int computeQueryQuota(ConnectionTrack conn, char *errorbuf, int errorbufsize)
 		 */
 		if ( conn->MinSegCountFixed != conn->MaxSegCountFixed )
 		{
-			vseglimit = conn->VSegLimit ?
-						conn->VSegLimit :
-						conn->VSegLimitPerSeg * PRESPOOL->AvailNodeCount;
+			vseglimit = conn->VSegLimitPerSeg * PRESPOOL->AvailNodeCount;
```
Comment (huor) [inline src/backend/resourcemanager/resqueuemanager.c]: I would be clearer using below:
vseglimit = min(conn->VsegLimit, (conn->VSegLimitPerSeg) * (PRESPOOL->AvailNodeCount));

Comment (jiny2) [inline src/backend/resourcemanager/resqueuemanager.c]: I'd like to keep original code.

**Comment by yaoj2** [comment]
+1

**Pull request closed without merging**